docs/interfaces.md

```diff
@@ -0,0 +1,592 @@
+# FPC Component Interfaces and States
```

Comment:
BTW: shouldn't we put this file into `docs/design/fabric-v2+` as the rest of the design docs?

Reply:
It seems appropriate.

Reply:
moved!